library/network_connections.py

```diff
@@ -107,15 +112,11 @@ def _link_read_address(ifname):
 
     @staticmethod
     def _link_read_permaddress(ifname):
-        try:
-            out = Util.check_output(["ethtool", "-P", ifname])
-        except MyError:
-            return None
+        perm_addr = ethtool.get_perm_addr(ifname)
+        if perm_addr:
+            perm_addr = Util.mac_norm(perm_addr)
```

Comment:
the normalization is not needed anymore, since ethtool already returns the normalized form.

Reply:
Removed, thanks :-)